ext/standard/array.c

```diff
@@ -3861,6 +3915,10 @@ static zend_always_inline void php_array_replace_wrapper(INTERNAL_FUNCTION_PARAM
 			zend_hash_merge(dest, Z_ARRVAL_P(arg), zval_add_ref, 1);
 		}
 	}
+
+	if (in_place) {
+		GC_ADDREF(dest);
+	}
```

Comment:
I assume, you need this to avoid destruction in zend_vm_stack_free_args().
You may also try ``ZVAL_NULL()`` or ``ZVAL_UNDEF()`` of the arg here or in the first place (instead of setting ``in_place``). This may only change the argument value in backtrace.

ADDREF is safer.